posydon/binary_evol/SN/step_SN.py

```diff
@@ -1624,19 +1627,44 @@ def SNCheck(
             # update the binary object which was bound at least before the SN
             if flag_binary:
                 # update the tilt
-                if binary.event == "CC1":
-                    binary.star_1.spin_orbit_tilt = tilt
-                elif binary.event == "CC2":
-                    binary.star_2.spin_orbit_tilt = tilt
-                else:
-                    raise ValueError("This should never happen!")
-
-                # compute new orbital period before reseting the binary properties
+                #Check if this is the first SN
+                first_SN = binary.true_anomaly_first_SN is None
 
                 for key in BINARYPROPERTIES:
-                    if key != 'nearest_neighbour_distance':
+                    if key != 'nearest_neighbour_distance' and key != 'true_anomaly_first_SN':
                         setattr(binary, key, None)
 
+                if first_SN:
+                    # update the tilt
+                    binary.star_1.spin_orbit_tilt_first_SN = tilt
+                    binary.star_2.spin_orbit_tilt_first_SN = tilt
+                    binary.true_anomaly_first_SN = true_anomaly
+                else:
+                    if binary.event == 'CC2':
+                        # Assume progenitor has aligned with the preSN orbital angular momentum
+                        binary.star_2.spin_orbit_tilt_second_SN = tilt
+                        binary.star_1.spin_orbit_tilt_second_SN = get_combined_tilt(
+                            tilt_1 = binary.star_1.spin_orbit_tilt_first_SN, 
+                            tilt_2 = tilt, 
+                            true_anomaly_1 = binary.true_anomaly_first_SN, 
+                            true_anomaly_2 = true_anomaly
+                            )
+                        binary.true_anomaly_SN2 = true_anomaly
+                    elif binary.event == 'CC1':
+                        # Assume progenitor has aligned with the preSN orbital angular momentum
+                        binary.star_1.spin_orbit_tilt_second_SN = tilt
+                        binary.star_2.spin_orbit_tilt_second_SN = get_combined_tilt(
+                            tilt_1 = binary.star_1.spin_orbit_tilt_first_SN, 
```

Comment:
Shouldn't this be `binary.star_2.spin_orbit_tilt_first_SN`.
In the current version, the tilt of the first SN is the same for both stars, hence it won't change the values.

Reply:
@mkruckow Thanks for catching this. Fixed.